src/averageAges.js

```diff
@@ -55,7 +82,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  // Всі жінки у яких є діти
+  const women = people.filter((woman) => {
+    return woman.sex === 'f'
+      && people.find((person) => person.mother === woman.name);
+  });
+
+  const diffs = women.map((woman) => {
+    // Всі діти конкретної жінки, або тільки сини
+    const children = people.filter((person) => {
+      return !onlyWithSon
+        ? person.mother === woman.name
+        : person.mother === woman.name && person.sex === 'm';
+    });
+
+    // Повертаємо вік народження дитини
```

Comment:
Again, remove the comment on line 60 for better readability and use English for future comments. 🌐